src/metabase/api/card.clj

```diff
@@ -188,10 +189,24 @@
       (qp.async/result-metadata-for-query-async query))))
 
 (defn check-data-permissions-for-query
-  "Check that we have *data* permissions to run the QUERY in question."
+  "Make sure the Current User has the appropriate *data* permissions to run `query`. We don't want Users saving Cards
+  with queries they wouldn't be allowed to run!"
   [query]
   {:pre [(map? query)]}
-  (api/check-403 (query-perms/can-run-query? query)))
+  (when-not (query-perms/can-run-query? query)
+    (let [required-perms (try
+                           (query-perms/perms-set query :throw-exceptions? true)
+                           (catch Throwable e
+                             e))]
+      (throw (ex-info (tru "You cannot save this Question because you do not permissions to run its query.")
```

Comment:
nit: "you do not have permissions" (missing "have")

Reply:
Oops. Don't know how I missed that 😢